apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/UtilsController.java

```diff
@@ -0,0 +1,521 @@
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.info.Info;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Encoding;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.resteasy.annotations.Form;
+import org.jboss.resteasy.annotations.providers.multipart.MultipartForm;
+import org.jboss.resteasy.plugins.providers.multipart.InputPart;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataInput;
+
+/**
+ * Expose several utility methods
+ */
+@NoArgsConstructor
+@Slf4j
+@Path("/utils")
+@OpenAPIDefinition(info = @Info(description = "Provide several utility methods"))
+@Singleton
+public class UtilsController {
+
+    private BlockchainConfig blockchainConfig;
+
+    @Inject
+    public UtilsController(BlockchainConfig blockchainConfig) {
+        this.blockchainConfig = Objects.requireNonNull(blockchainConfig);
+    }
+
+    @Path("/qrcode/encode")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Operation(
+        summary = "The API converts a UTF-8 string to a base64-encoded jpeg image of a 2-D QR (Quick Response) code",
+        description = "The output qrCodeBase64 string can be incorporated into an in-line HTML image like this: &lt;img src=\"data:image/jpeg;base64,qrCodeBase64\"&gt; ",
+        tags = {"utility"},
+        responses = {
+            @ApiResponse(responseCode = "200", description = "Successful execution",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = QrEncodeDto.class)))
+        })
+    @PermitAll
+    public Response encodeQRCode(
+        @Parameter(name = "qrCodeData", description = "QR code data", required = true) // works on UI without good description
+            @FormParam("qrCodeData") String qrCodeData,
+        @Parameter(name = "width", description = "QR code image width, optional") // works on UI without good description
+            @FormParam("width") @DefaultValue("0") String widthStr,
+        @Parameter(name = "height", description = "QR code image height, optional") // works on UI without good description
+            @FormParam("height") @DefaultValue("0") String heightStr
+    ) {
+        log.debug("Started encodeQRCode: \n\t\twidthStr={}, heightStr={}, qrCodeData={}", widthStr, heightStr, qrCodeData);
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        if (StringUtils.isEmpty(qrCodeData)) {
```

Comment:
I think that better to add @NotBlank annotation for string parameters to get rid off such boilerplate code